src/clients/Deluge.ts

```diff
@@ -0,0 +1,236 @@
+import axios, { AxiosResponse } from "axios";
+import { InjectionResult } from "../constants.js";
+import { CrossSeedError } from "../errors.js";
+import { Label, logger } from "../logger.js";
+import { Metafile } from "../parseTorrent.js";
+import { getRuntimeConfig } from "../runtimeConfig.js";
+import { Searchee } from "../searchee.js";
+import { TorrentClient } from "./TorrentClient.js";
+
+export default class Deluge implements TorrentClient {
+	private msgId = 0;
+	private loggedIn = false;
+	private delugeCookie = "";
+	private delugeWebUrl: URL;
+	private delugeLabel: string;
+	private enLabeled: Promise<boolean>;
+
+	constructor() {
+		this.delugeWebUrl = new URL(`${getRuntimeConfig().delugeWebUrl}`);
+		this.delugeLabel = getRuntimeConfig().dataCategory;
+	}
+
+	/**
+	 * validates the login and host for deluge webui
+	 */
+	async validateConfig(): Promise<void> {
+		const url = new URL(this.delugeWebUrl);
+		if (url.username && !url.password) {
+			throw new CrossSeedError(
+				"you need to define a password in the delugeWebUrl. (eg: http://:<PASSWORD>@localhost:8112)"
+			);
+		}
+		const result = await this.authenticate();
+		if (!result) {
+			logger.verbose({
+				label: Label.DELUGE,
+				message: "failed to authenticate the deluge daemon.",
+			});
+			throw new CrossSeedError(
+				`failed to authenticate the deluge daemon.`
+			);
+		} else {
+			this.enLabeled = this.labelEnabled();
+		}
+	}
+
+	/**
+	 * connects and authenticates to the webui
+	 */
+	private async authenticate(): Promise<boolean> {
+		if (!this.loggedIn) {
```

Comment:
I have a feeling that `this.loggedIn` is going to fall out of date, either if the deluge server restarts while cross-seed is running, or if we get a login timeout, etc. 

Can we instead assume that we're always logged in, but have login-and-retry logic if we get an auth error during any call? feel free to take a look at the qbit logic:

 https://github.com/cross-seed/cross-seed/blob/c012a088596fac654cc38ad8ce9bd89154f85bf7/src/clients/QBittorrent.ts#L160-L167

Reply:
Implemented. Will be in next commit. 

Just double check my logic conforms to intended behaviors.